src/download.c

```diff
@@ -500,6 +500,12 @@ static int poll_fewer_than(size_t xfer_queue_high, size_t xfer_queue_low)
 			}
 		}
 
+		// Do more work before processing the queue
+		curlm_ret = curl_multi_perform(mcurl, &running);
```

Comment:
Good catch in the docs, but that error is only returned in curl < 7.20.0. Clear Linux runs a much newer version.